File: src/fahamu/provider/Resources.java
package fahamu.provider;

import java.io.File;
import java.net.MalformedURLException;
import java.net.URL;
import java.nio.file.Path;
import java.nio.file.Paths;

public class Resources {

    public String ROOT_PATH_FOR_RESOURCE=""; //empty string solve a nul bug
    public String BUSINESS_NAME = "SMART STOCK MANAGER";
    public final Path SERVER_CREDENTIAL_FILE = getResourceAsPath("res/sqlite3/serverCredential.db.encrypted");
    public final URL BUSINESS_LOGO =getResourceAsUrl("res/image/lbLogo.jpg");
    public final URL CART_ICON=getResourceAsUrl("res/image/cart.png");
    public final URL DATABASE_ERROR_ICON=getResourceAsUrl("res/image/databaseIcon.png");
    public final URL ICON = getResourceAsUrl("res/image/ssmlogo.png");
    public final URL BACKGROUND_IMAGE_PATH = getResourceAsUrl("res/image/calculate.jpg");
    public final URL WRONG_PASSWORD_ICON=getResourceAsUrl("res/image/manicon.png");
    public final URL LOG_IN_UI_FXML = getResourceAsUrl("res/fxmls/loginUi.fxml");
    public final URL LEFT_DRAWER_LAYOUT_FXML = getResourceAsUrl("res/fxmls/logInLeftDrawer.fxml");
    public final URL SALE_UI_LAYOUT_FXML = getResourceAsUrl("res/fxmls/sellerUi.fxml");
    public final URL LOG_IN_BANNER_IMAGE = getResourceAsUrl("res/image/googleSign.png");
    public final URL LOG_IN_DRAWER_STACK_PANE_CONTENT =getResourceAsUrl("res/fxmls/logInDrawerContent.fxml");
    public final URL SIGN_UP_BANNER_IMAGE=getResourceAsUrl("res/image/googleSignUp.png");
    public final URL SETTING_BANNER_IMAGE=getResourceAsUrl("res/image/setting.png");

    public final String ADMIN = "admin";
    public byte[] SERVER_IPA4_ADDRESS = new byte[]{(byte) 192, (byte) 168, 0, 2};

    public Resources() {

    }

    private URL getResourceAsUrl(String name) {
        Path path = Paths.get(ROOT_PATH_FOR_RESOURCE, name);
        File file = path.toFile();
        try {
            return file.toURI().toURL();
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
        return null;
    }

    private Path getResourceAsPath(String name) {
        //System.out.println(Paths.get(rootResourcePath,name));
        return Paths.get(ROOT_PATH_FOR_RESOURCE, name);
    }
}

File: src/main/java/fahamu/provider/Resources.kt
//package fahamu.provider
//
//import java.net.URL
//
//open class Resources() {
//
//    var ROOT_PATH_FOR_RESOURCE = "" //empty string solve a nul bug
//    //public final Path SERVER_CREDENTIAL_FILE = getResourceAsPath("../sqlite3/serverCredential.db.encrypted");
//    val BUSINESS_LOGO = getResourceAsUrl("image/lbLogo.jpg")
//    val CART_ICON = getResourceAsUrl("image/cart.png")
//    val DATABASE_ERROR_ICON = getResourceAsUrl("image/databaseIcon.png")
//    val ICON = getResourceAsUrl("image/ssmlogo.png")
//    val BACKGROUND_IMAGE_PATH = getResourceAsUrl("image/calculate.jpg")
//    val wrongPasswordIcon = getResourceAsUrl("image/manicon.png")
//    val LOG_IN_UI_FXML = getResourceAsUrl("fxmls/loginUi.fxml")
//    val LEFT_DRAWER_LAYOUT_FXML = getResourceAsUrl("fxmls/logInLeftDrawer.fxml")
//    val SALE_UI_LAYOUT_FXML = getResourceAsUrl("fxmls/sellerUi.fxml")
//    val LOG_IN_BANNER_IMAGE = getResourceAsUrl("image/googleSign.png")
//    val LOG_IN_DRAWER_STACK_PANE_CONTENT = getResourceAsUrl("fxmls/logInDrawerContent.fxml")
//    val SIGN_UP_BANNER_IMAGE = getResourceAsUrl("image/googleSignUp.png")
//    val SETTING_BANNER_IMAGE = getResourceAsUrl("image/setting.png")
//
//    val ADMIN = "admin"
//
//    companion object {
//        val BUSINESS_NAME = "SMART STOCK MANAGER"
//        var SERVER_IPA4_ADDRESS = byteArrayOf(192.toByte(), 168.toByte(), 0, 2)
//
//        fun  getResourceAsUrl(name: String): URL {
//            return Thread.currentThread().contextClassLoader.getResource(name)
//        }
//    }
//
//}
